presto-product-tests/src/main/java/io/prestosql/tests/hive/acid/TestFlatTableConvertedToTransactionalTable.java

```diff
@@ -0,0 +1,119 @@
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive.acid;
+
+import io.prestosql.tempto.query.QueryResult;
+import io.prestosql.tests.hive.HiveProductTest;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.TestGroups.HIVE_TRANSACTIONAL;
+import static io.prestosql.tests.TestGroups.STORAGE_FORMATS;
+import static io.prestosql.tests.utils.QueryExecutors.onHive;
+
+public class TestFlatTableConvertedToTransactionalTable
+        extends HiveProductTest
+{
+    @Test(groups = {STORAGE_FORMATS, HIVE_TRANSACTIONAL})
+    public void testReadingFullAcidConvertedTable()
+    {
+        if (getHiveVersionMajor() < 3) {
+            throw new SkipException("Presto Hive transactional tables are supported with Hive version 3 or above");
+        }
+
+        String tableNameInDatabase = "reading_full_acid_converted_table";
+
+        String createTable = "CREATE TABLE IF NOT EXISTS " +
+                tableNameInDatabase +
+                " (col INT, fcol INT) " +
+                "STORED AS ORC TBLPROPERTIES ('transactional'='false')";
+        onHive().executeQuery(createTable);
+
+        try {
+            onHive().executeQuery(
+                    "INSERT OVERWRITE TABLE " + tableNameInDatabase + " select 1, 11");
+
+            String selectAllFromTable = "SELECT * FROM " + tableNameInDatabase;
+            QueryResult onePartitionQueryResult = query(selectAllFromTable);
+            assertThat(onePartitionQueryResult).containsOnly(row(1, 11));
+
+            onHive().executeQuery(
+                    "INSERT INTO TABLE " + tableNameInDatabase + " select 2, 22");
+            onePartitionQueryResult = query(selectAllFromTable);
+            assertThat(onePartitionQueryResult).hasRowsCount(2);
+
+            onHive().executeQuery(
+                    "ALTER TABLE  " + tableNameInDatabase + " SET TBLPROPERTIES ('transactional'='true')");
+
+            // delete a row
+            onHive().executeQuery(
+                    "DELETE FROM " + tableNameInDatabase + " WHERE fcol=22");
+            onePartitionQueryResult = query(selectAllFromTable);
+            assertThat(onePartitionQueryResult).containsOnly(row(1, 11));
+            // update the existing row
+            onHive().executeQuery(
+                    "UPDATE " + tableNameInDatabase + " set col = 3 where fcol=11");
+            onePartitionQueryResult = query(selectAllFromTable);
+            assertThat(onePartitionQueryResult).containsOnly(row(3, 11));
+        }
+        finally {
+            onHive().executeQuery("DROP TABLE " + tableNameInDatabase);
+        }
+    }
+
+    @Test(groups = {STORAGE_FORMATS, HIVE_TRANSACTIONAL})
+    public void testReadingInsertOnlyAcidConvertedTable()
```

Comment:
call it `testReadInsertOnlyWithOriginalFiles` and move to `TestHiveTransactionalTable`

add a variant with bucketed table

Reply:
fixed.